ProcessMaker/processmaker
# Implement Saving Screen Templates Feature

This PR implements both frontend and backend functionalities for saving a screen template. This feature allows users to select the 'Save as Template' option from the screen listing option menu. Upon selection, a new 'Create Template' modal appears, allowing the user to configure various settings such as name, description, version, thumbnails, category, and public settings for the template. Upon saving, the template is created, and a success message is displayed to the user.

## Solution
- Added and refactored existing frontend components and UI elements to faciliate template creation
- Implemented backend logic to handle saving screen templates with specificed configurations
- integrated frontend and backend components. 

## How to Test

**Test Case A**

1. Navigate to the Screen listing page.
2. Select the 'Save as Template' option from the options menu of a screen
3. Fill in the details in the 'Create Template' modal.
4. Click the 'Save' button
5. Verify that the template is created successfully
6. Confirm that a success message is displayed to indicate successful template creation.
7. Verify the created template in 'My Templates' or 'Public Templates' tab.

**Test Case B**

1. Navigate to the Screen listing page.
2. Select the 'Save as Template' option from the options menu of a screen
3. Fill in the details in the 'Create Template' modal.
4. Leave some required data empty
5. Verify the 'Save' button is disabled.
6. Add in missing data
7. Verify the 'Save' button is enabled
8. Select the 'Save' button
9. Confirm that a success message is displayed to indicate successful template creation.
10. Verify the created template in 'My Templates' or 'Public Templates' tab.


**Test Case C**

1. Navigate to the Screen listing page.
2. Select the 'Save as Template' option from the options menu of a screen
3. Fill in the details in the 'Create Template' modal.
4. Ensure the name of the template is the same as an existing template.
5. Select the 'Save' button
6. Ensure a Existing template warning appears
7. Verify that both 'Update' and 'Save' buttons work as expected

## Related Tickets & Packages
- [FOUR-13538](https://processmaker.atlassian.net/browse/FOUR-13538)

ci:next
ci:deploy

## Code Review Checklist


[FOUR-13538]: https://processmaker.atlassian.net/browse/FOUR-13538?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
